main.go

```diff
@@ -381,18 +430,57 @@ func runCLIApp(c *cli.Context) (err error) {
 	{
 		mfs, err = mountWithArgs(bucketName, mountPoint, flags, mountConfig)
 
-		if err == nil {
+		// This utility is to absorb the error
+		// returned by daemonize.SignalOutcome calls by simply
+		// logging them as error logs.
+		callDaemonizeSignalOutcome := func(err error) {
+			if err2 := daemonize.SignalOutcome(err); err2 != nil {
+				logger.Errorf("Failed to signal to daemon: %v", err2)
+			}
+		}
+
+		markSuccessfullMount := func() {
 			// Print the success message in the log-file/stdout depending on what the logger is set to.
 			logger.Info(SuccessfulMountMessage)
+			callDaemonizeSignalOutcome(nil)
+		}
 
-			daemonize.SignalOutcome(nil)
+		if err == nil {
+			if isDynamicMount {
+				markSuccessfullMount()
+			} else {
+				switch flags.MetadataPrefetchMode {
+				case config.MetadataPrefetchModeDisabled:
+					markSuccessfullMount()
+				case config.MetadataPrefetchModeSynchronous:
+					if err = callListRecursive(mountPoint); err != nil {
+						// Printing via mountStatus will have duplicate logs on the console while
+						// mounting gcsfuse in foreground mode. But this is important to avoid
+						// losing error logs when run in the background mode.
+						logger.Errorf("%s: %v\n", UnsuccessfulMountMessagePrefix, err)
```

Comment:
Use %w for error?

Reply:
%w only works for fmt.Errorf for wrapping errors to creating more complex errors, not for actually print statements.